pkg/api/aim/query/query.go

```diff
@@ -581,6 +612,57 @@ func (pq *parsedQuery) parseName(node *ast.Name) (any, error) {
 					}
 				},
 			), nil
+		case "re":
+			return callable(
+				func(args []ast.Expr) (any, error) {
+					operation, ok := args[0].(*ast.Str)
+					if !ok {
+						return nil, errors.New("re operation type has to be ast.Str")
+					}
+					switch operation.S {
+					case RegexpOperationMatch:
+						fallthrough
+					case RegexpOperationSearch:
+						// subtract 1, because first argument is an operation argument.
+						if len(args)-1 != 2 {
+							return nil, errors.New("re.match function support exactly 2 arguments")
+						}
+
+						regexp, ok := args[1].(*ast.Str)
+						if !ok {
+							return nil, errors.New("first argument type for re.match function has to be ast.Str")
+						}
+						attribute, ok := args[2].(*ast.Attribute)
+						if !ok {
+							return nil, errors.New("second argument type for re.match function has to be ast.Attribute")
+						}
+
+						attributes, err := pq.parseName(attribute.Value.(*ast.Name))
```

Comment:
you should never call `parseName` directly but instead use `parseNode` which will help with proper error handling

Reply:
done.